src/figuresClasses.ts

```diff
@@ -1,11 +1,91 @@
-export interface Figure {}
+/* eslint-disable @typescript-eslint/lines-between-class-members */
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: 'red' | 'green' | 'blue';
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = 'triangle';
+  color: 'red' | 'green' | 'blue';
+  private a: number;
+  private b: number;
+  private c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    this.color = color;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than 0');
+    }
+
+    const sides = [a, b, c].sort((x, z) => z - x);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error('This triangle cannot exist');
+    }
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
why?
i think it's extra

Reply:
nope, they are necessary, i wouldn't be able to pass the variables `a`, `b` and `c` to other class methods, such as `getArea()`.

Reply:
> nope, they are necessary, i wouldn't be able to pass the variables `a`, `b` and `c` to other class methods, such as `getArea()`.
Ohhh, yes, I messed up.
In order to access variables without - `this`, u need to initialize the public variable in the constructor.
Then you won't need to create it at the beginning and assign them at the end. 
Iєll show how it will look in a separate comment